### PR TITLE
Added user_for_token method, moved r check into its own function

### PR DESCRIPTION
**Pull request by DeepHorizons**
This pull request adds the `user_for_token` method to the HubAuth class. This enables easy access to authenticate and get a user by their token.

The response check was moved into its own function to remove duplicating code.

### CODE REVIEW

File: jupyterhub/services/auth.py [hunk 1]
```diff
@@ -138,6 +138,27 @@ class HubAuth(Configurable):
...
+            app_log.error("I don't have permission to verify cookies, my auth token may have expired: [%i] %s", r.status_code, r.reason)
+            raise HTTPError(500, "Permission failure checking authorization, I may need a new token")
+        elif r.status_code >= 500:
+            app_log.error("Upstream failure verifying auth token: [%i] %s", r.status_code, r.reason)
+            raise HTTPError(502, "Failed to check authorization (upstream problem)")
+        elif r.status_code >= 400:
+            app_log.warning("Failed to check authorization: [%i] %s", r.status_code, r.reason)
+            raise HTTPError(500, "Failed to check authorization")
+        else:
+            app_log.debug("Received request from Hub user %s", data)
```
Comment (minrk) [inline jupyterhub/services/auth.py]: `data` here isn't declared. Perhaps:

```python
data = r.json()
app_log.debug(...)
return data
```

File: jupyterhub/services/auth.py [hunk 2]
```diff
@@ -138,6 +138,27 @@ class HubAuth(Configurable):
...
+    def _check_response_for_authorization(self, r):
+        """Verify the response for authorizing a user
+
+        Raises an error if the response failed, otherwise returns the json
+        """
+        if r.status_code == 404:
+            app_log.warning("No Hub user identified for request")
+            data = None
+        elif r.status_code == 403:
+            app_log.error("I don't have permission to verify cookies, my auth token may have expired: [%i] %s", r.status_code, r.reason)
```
Comment (minrk) [inline jupyterhub/services/auth.py]: Update message to be more general:

"I don't have permission to check authorization with JupyterHub; my auth..."

**Comment by DeepHorizons** [comment]
@minrk Fixed the message and the logging error.

**Comment by minrk** [comment]
Great, thanks!

**Pull request merged**